### PR TITLE
fix 'channels_first' conv3d with None shape inputs

### PR DESCRIPTION
**Pull request by shiyangc-intusurg**
use array_ops.shape such that 'channels_first' conv3d can support None shape inputs.

### CODE REVIEW

**Comment by shiyangc-intusurg** [comment]
I signed it!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 46 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by fchollet — CHANGES_REQUESTED** [review]
Please add unit tests for this change (it is supposed to fix an issue, so the unit test should make it clear what the issue was). Thank you

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 32 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Lucie2A** [comment]
Here is a code snippet to reproduce the bug and test the proposed fix: 

```
import tensorflow as tf
x = tf.placeholder(dtype=tf.float32, shape=[None, 1, None, None, None])
y = tf.layers.conv3d(inputs=x, filters=32, kernel_size=3, data_format='channels_first')
```

**Comment by Lucie2A** [comment]
With the proposed fix, I can confirm that conv3d can be used with None shape inputs.

**System information**
- Have I written custom code: yes, as there is no TF sample using conv3d with None shape inputs
- OS Platform and Distribution: Linux Ubuntu 16.04.5 LTS x86_64
- Mobile device: N/C
- TensorFlow installed from: source
- TensorFlow version: b'v1.12.0-874-g70b346d' 1.12.0 / b'v1.10.1-0-g4dcfddc' 1.10.1 / also tested on v1.11
- Python version: 3.5
- Bazel version: 0.15.0 / 0.16.1
- GCC/Compiler version: 5.4.0
- CUDA/cuDNN version: rocm 1.9.2 / cuda 9.2 / cudnn 7.1.4
- GPU model and memory: AMD RX VEGA 56 - 8Gb / Nvidia GTX 1060 - 6 Gb

Of course conv3d work as expected only with Nvidia GPU, because with AMD GPU there is the MIOpen layout that is limited to 4D tensors and you need a 5D tensors to run conv3d.

**Comment by rthadur** [comment]
@shiyangc-intusurg can you please resolve conflicts ?

**Comment by shiyangc-intusurg** [comment]
It seems since 2.0.0alpha, conv3d does not need to be treat specially when doing bias_add. That means it should not have any problem doing conv3d with None shape and channel_first. If you can verify that is true, feel free to close this ticket.

**Comment by tanzhenyu** [comment]
From the context looks like conv3d is fixing this issue. Closing it.
